subprojects/packagefiles/opus/0001-meson-Fix-symbol-leaks-on-Windows.patch

```diff
@@ -0,0 +1,75 @@
+From 5bdcf4afa6ed37ee449ba2cacffca8582e1fb377 Mon Sep 17 00:00:00 2001
+From: "L. E. Segovia" <amy@centricular.com>
+Date: Sun, 6 Jul 2025 13:05:29 -0300
+Subject: [PATCH] meson: Fix symbol leaks on Windows
+
+---
+ celt/meson.build | 2 +-
+ meson.build      | 2 +-
+ silk/meson.build | 2 +-
+ src/meson.build  | 6 +++++-
+ 4 files changed, 8 insertions(+), 4 deletions(-)
+
+diff --git a/celt/meson.build b/celt/meson.build
+index 370ea1fe..6d437901 100644
+--- a/celt/meson.build
++++ b/celt/meson.build
+@@ -50,7 +50,7 @@ if host_cpu_family in ['arm', 'aarch64'] and have_arm_intrinsics_or_asm
+ endif
+ 
+ celt_c_args = []
+-if host_system == 'windows'
++if host_machine.system() == 'windows' and get_option('default_library') == 'shared'
+   celt_c_args += ['-DDLL_EXPORT']
+ endif
+ 
+diff --git a/meson.build b/meson.build
+index 41f69353..254c9fe4 100644
+--- a/meson.build
++++ b/meson.build
+@@ -1,6 +1,6 @@
+ project('opus', 'c',
+   version: run_command('meson/get-version.py', '--package-version', check: true).stdout().strip(),
+-  meson_version: '>=0.54.0',
++  meson_version: '>=1.3.0',
+   default_options: ['warning_level=2',
+                     'c_std=gnu99',
+                     'buildtype=debugoptimized'])
+diff --git a/silk/meson.build b/silk/meson.build
+index 70692372..59eaa16d 100644
+--- a/silk/meson.build
++++ b/silk/meson.build
+@@ -40,7 +40,7 @@ foreach intr_name : ['sse4_1', 'neon_intr']
+ endforeach
+ 
+ silk_c_args = []
+-if host_machine.system() == 'windows'
++if host_machine.system() == 'windows' and get_option('default_library') == 'shared'
+   silk_c_args += ['-DDLL_EXPORT']
+ endif
+ 
+diff --git a/src/meson.build b/src/meson.build
+index cc07ff06..bc0c5e51 100644
+--- a/src/meson.build
++++ b/src/meson.build
+@@ -15,12 +15,16 @@ opus_lib = library('opus',
+   opus_sources,
+   version: libversion,
+   darwin_versions: macosversion,
+-  c_args: opus_lib_c_args,
++  c_shared_args: opus_lib_c_args,
+   include_directories: opus_includes,
+   link_with: [celt_lib, silk_lib],
+   dependencies: libm,
+   install: true)
+ 
++if get_option('default_library') == 'both'
++  opus_lib = opus_lib.get_static_library()
++endif
```

Comment:
Why the static lib when orc uses the shared lib when default_library==both?

Reply:
I wasn't particularly sure of which to use or the convention underlying each of them; in either case, the libraries are built statically.

Reply:
Meson's convention is to prefer shared libraries when both are enabled, so let's do that.